packages/database/src/core/SyncPoint.ts

```diff
@@ -237,71 +222,57 @@ export class SyncPoint {
     return { removed: removed, events: cancelEvents };
   }
 
-  /**
-   * @return {!Array.<!View>}
-   */
   getQueryViews(): View[] {
-    const values = Object.keys(this.views_).map(key => this.views_[key]);
-    return values.filter(function(view) {
-      return !view
-        .getQuery()
-        .getQueryParams()
-        .loadsAllData();
-    });
+    return Array.from(this.views.values()).filter(
+      view =>
+        !view
+          .getQuery()
+          .getQueryParams()
+          .loadsAllData()
+    );
   }
 
   /**
-   *
-   * @param {!Path} path The path to the desired complete snapshot
-   * @return {?Node} A complete cache, if it exists
+   * @param path The path to the desired complete snapshot
+   * @return A complete cache, if it exists
    */
   getCompleteServerCache(path: Path): Node | null {
     let serverCache: Node | null = null;
-    forEach(this.views_, (key: string, view: View) => {
+    for (const view of this.views.values()) {
```

Comment:
This creates a copy.

Reply:
`this.views` is a `Map`. This returns an iterator, not a copy.